src/node_url.cc

```diff
@@ -17,9 +17,20 @@
 
 #if defined(NODE_HAVE_I18N_SUPPORT)
 #include <unicode/utf8.h>
+#include <unicode/utf16.h>
 #include <unicode/utf.h>
 #endif
 
+#if !defined(U16_IS_SURROGATE)
+#define U16_IS_LEAD(c) (((c) & 0xFC00) == 0xD800)
+#define U16_IS_TRAIL(c) (((c) & 0xFC00) == 0xDC00)
+#define U16_IS_SURROGATE(c) (((c) & 0xF800) == 0xD800)
+#define U16_IS_SURROGATE_LEAD(c) (((c) & 0x400) == 0)
+#define U16_IS_SURROGATE_TRAIL(c) (((c) & 0x400) != 0)
+#endif
```

Comment:
I'd turn these into inline functions.  It adds an extra bit of type checking and avoids possible divergence between intl and non-intl builds.